toponetx/algorithms/distance_measures.py

```diff
@@ -0,0 +1,196 @@
+from toponetx.classes.combinatorial_complex import CombinatorialComplex
+from toponetx.classes.simplicial_complex import SimplicialComplex
+from toponetx.exception import TopoNetXError
+
+__all__ = ["node_diameters", "cell_diameters", "diameter", "cell_diameter"]
+
+
+def node_diameters(complex):
+    """Return the node diameters of the connected components in cell complex.
+
+    Returns
+    -------
+    diameters : list
+        List of the diameters of the s-components.
+    components : list
+        List of the s-component nodes.
+
+    Example
+    -------
+    >>> CC = CellComplex()
+    >>> CC.add_cell([2,3,4],rank=2)
+    >>> CC.add_cell([5,6,7],rank=2)
+    >>> list(node_diameters(CC))
+    >>> CCC = CC.to_combinatorial_complex()
+    >>> list(node_diameters(CCC))
+    >>> CHG = CC.to_colored_hypergraph()
+    >>> list(node_diameters(CHG))
+    """
+    node_dict, A = complex.node_to_all_cell_adjacnecy_matrix(index=True)
+    if isinstance(complex, ColoredHyperGraph) and not isinstance(
+        complex, CombinatorialComplex
+    ):
+        node_dict = {v: k[0] for k, v in node_dict.items()}
+    else:
+        node_dict = {v: k for k, v in node_dict.items()}
+
+    G = nx.from_scipy_sparse_array(A)
+    diams = []
+    comps = []
+    for c in nx.connected_components(G):
+        diamc = nx.diameter(G.subgraph(c))
+        temp = set()
+        for e in c:
+            temp.add(node_dict[e])
+        comps.append(temp)
+        diams.append(diamc)
+    return diams, comps
+
+
+def cell_diameters(complex, s=1):
+    """Return the cell diameters of the s_cell_connected component subgraphs.
+
+    Parameters
+    ----------
+    complex : Complex
+        Supported complexes are cell/combintorial and hypegraphs.
+    s : int, optional
+
+    Returns
+    -------
+    list of diameters : list
+        List of cell_diameters for s-cell component subcomplexes in the cell complex.
+
+    list of component : list
+        List of the cell uids in the s-cell component subcomplexes.
+
+    Example
+    -------
+    >>> CC = CellComplex()
+    >>> CC.add_cell([2,3,4],rank=2)
+    >>> CC.add_cell([5,6,7],rank=2)
+    >>> list(cell_diameters(CC))
+    >>> CCC = CC.to_combinatorial_complex()
+    >>> list(cell_diameters(CCC))
+    >>> CHG = CC.to_colored_hypergraph()
+    >>> list(cell_diameters(CHG))
+    """
+    if not isinstance(complex, (CellComplex, CombinatorialComplex, ColoredHyperGraph)):
+        raise ValueError("Input complex {complex} is not supported.")
+    coldict, A = complex.all_cell_to_node_coadjacnecy_matrix(index=True)
+    coldict = {v: k for k, v in coldict.items()}
+
+    G = nx.from_scipy_sparse_array(A)
+    diams = []
+    comps = []
+    for c in nx.connected_components(G):
+        diamc = nx.diameter(G.subgraph(c))
+        temp = set()
+        for e in c:
+            temp.add(coldict[e])
+        comps.append(temp)
+        diams.append(diamc)
+    return diams, comps
+
+
+def diameter(complex) -> int:
```

Comment:
NumPy's documentation style dictates that this headline should read Examples even if there is only one (same throughout this file).